ttnn/cpp/ttnn/tensor/tensor.hpp

```diff
@@ -17,6 +17,7 @@
 #include "common/tt_backend_api_types.hpp"
 #include "ttnn/common/constants.hpp"
 #include "ttnn/tensor/types.hpp"
+#include "tt_metal/detail/tt_metal.hpp"
```

Comment:
could we just move these impls to the `tensor.cpp` and introduce the header dependency there?

Reply:
Moved include and dependent fns to cpp. Thanks

Reply:
thank you!